### PR TITLE
feature/SWITCH-Create-switch-component

### PR DESCRIPTION
**Pull request by Chernovolos**
- add form switch component;
- add the form switch to the test;
- add validation for form switch to the validationSchema;
- add formSwitch to the SignInPayload interface for the test controller;

### CODE REVIEW

File: src/components/form-switch.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  label?: string
+  skipTouch?: boolean
+}
+
+const FSwitch: React.FC<FSwitchProps> = props => {
+  const { name, skipTouch, label, ...attr } = props;
+  const [field, meta] = useField({ name });
+  const valid = (skipTouch || meta.touched) && !meta.error;
+  const invalid = (skipTouch || meta.touched) && !!meta.error;
+  return <><FormControlLabel
```
Comment (zanusilker) [inline src/components/form-switch.tsx:20]:
```suggestion
  return <>
  <FormControlLabel
```

File: src/layouts/app/test/index.tsx [hunk 2]
```diff
@@ -2,12 +2,15 @@
 import * as yup from 'yup';
 import { Grid } from '@mui/material';
 import { Formik, Form } from 'formik';
+import MoodIcon from '@mui/icons-material/Mood';
```
Comment (zanusilker) [inline src/layouts/app/test/index.tsx:5]:
```suggestion
import { ... as MoodIcon } from '@mui/icons-material';
```

File: src/components/form-switch.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    label={label}
+    control={<Switch {...attr} {...field} color={validationStyles(valid, invalid)}/>}/>
+  { invalid && <FormHelperText color="error" sx={{ color: 'error.main', m: 0, pb: 2 }}>
+    { meta.error }
+  </FormHelperText> }
+  </>;
```
[the remark this answers is not in the document]
Comment (Chernovolos) [reply src/components/form-switch.tsx:26]: > does it have fullWidth props?

Switch and FormControlLabel don't have full Width props

File: src/components/form-switch.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+const FSwitch: React.FC<FSwitchProps> = props => {
+  const { name, skipTouch, label, ...attr } = props;
+  const [field, meta] = useField({ name });
+  const valid = (skipTouch || meta.touched) && !meta.error;
+  const invalid = (skipTouch || meta.touched) && !!meta.error;
+  return <>
+    <FormControlLabel
+      label={label}
+      control={<Switch {...attr} {...field} color={validationStyles(valid, invalid)}/>}
```
Comment (zanusilker) [inline src/components/form-switch.tsx:23]:
```suggestion
      control={<Switch color={validationStyles(valid, invalid)} {...field}  {...attr} />}
```

File: src/components/form-switch.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import React, { memo } from 'react';
+import { FormControlLabel, FormHelperText, Switch, SwitchProps } from '@mui/material';
+
+// local dependencies
+import { validationStyles } from './forms/helpers';
+
+interface FSwitchProps extends SwitchProps {
+  name: string
+  skipTouch?: boolean
+  label?: React.ReactNode
```
Comment (zanusilker) [inline src/components/form-switch.tsx:12]:
```suggestion
```

**Review by zanusilker — APPROVED** [review]

**Pull request merged**